integrations/transportation/minvolkswagen.js

```diff
@@ -0,0 +1,121 @@
+import request from 'superagent';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_CAR } from '../../definitions';
+import { HTTPError } from '../utils/errors';
+​
+async function loginWithPassword(email, password, logger) {
+  const res = await request
+    .post('https://auth-api.connectedcars.io/auth/login/email/password')
+    .set('x-organization-namespace', 'semler:minvolkswagen')
+    .send({ email, password });
+​
+  if (!res.ok) {
+    const text = await res.text();
+    throw new HTTPError(text, res.status);
+  }
+​
+  return res.body.token;
+}
+​
+async function loginWithDeviceToken(deviceToken, logger) {
+  const res = await request
+    .post('https://auth-api.connectedcars.io/auth/login/deviceToken')
+    .set('x-organization-namespace', 'semler:minvolkswagen')
+    .send({ deviceToken });
+​
+  if (!res.ok) {
+    const text = await res.text();
+    throw new HTTPError(text, res.status);
+  }
+​
+  return res.body.token;
+}
+​
+async function registerDevice(token, logger) {
+  const res = await request
+    .post('https://auth-api.connectedcars.io/user/registerDevice')
+    .set('x-organization-namespace', 'semler:minvolkswagen')
+    .set('Authorization', `Bearer ${token}`)
+    .send({
+      deviceModel: 'iPhone 8',
+      deviceName: 'Unknown',
+    });
+​
+  if (!res.ok) {
+    const text = await res.text();
+    throw new HTTPError(text, res.status);
+  }
+​
+  return res.body.deviceToken;
+}
+​
+async function connect(requestLogin, requestWebView, logger) {
+  const { username, password } = await requestLogin();
+​
+  const token = await loginWithPassword(username, password, logger);
+  const deviceToken = await registerDevice(token, logger);
+​
+  return { deviceToken };
+}
+​
+async function disconnect() {
+  return {};
+}
+​
+async function collect(state, logger) {
+  const { deviceToken } = state;
+​
+  const token = await loginWithDeviceToken(deviceToken, logger);
+​
+  const res = await request
+    .post('https://api.connectedcars.io/graphql?operationName=Trips')
+    .set('x-organization-namespace', 'semler:minvolkswagen')
+    .set('Authorization', `Bearer ${token}`)
+    .send({
+      operationName: 'Trips',
+      query:
+        'query Trips($vehicleId: ID!, $first: Int, $last: Int, $before: Cursor, $after: Cursor, $fromTime: Date, $toTime: Date, $ignoreEmpty: Boolean, $orderBy: TripOrder, $tripType: TripTypeFilter, $hasNote: Boolean) {vehicle(id: $vehicleId) {id trips(first: $first, last: $last, before: $before, after: $after, fromTime: $fromTime, toTime: $toTime, ignoreEmpty: $ignoreEmpty, orderBy: $orderBy, tripType: $tripType, hasNote: $hasNote) { items { id duration mileage startLatitude startLongitude endLatitude endLongitude time } pageInfo { hasNextPage hasPreviousPage startCursor endCursor __typename } __typename } __typename } }',
+      variables: {
+        first: 100,
```

Comment:
is it possible to fetch only items changed since last fetch? You could store the last fetch time in the `state`.
This will make the system more efficient.